platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/PathUtils.java

```diff
@@ -21,11 +23,104 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
 import org.apache.commons.lang.Validate;
+import org.codice.ddf.migration.MigrationReport;
+import org.codice.ddf.migration.MigrationWarning;
 
 public class PathUtils {
 
+  private static final String INVALID_NULL_PATH = "invalid null path";
+
+  /**
+   * Quietly deletes the specified file or recursively deletes the specified directory and audit the
+   * result. The provided directory is deleted.
+   *
+   * @param path the file or directory to be deleted
+   * @param type a string representing the type of file or directory to delete
+   * @return <code>true</code> if the file or directory was deleted, <code>false</code> otherwise
+   * @throws IllegalArgumentException if <code>path</code> or <code>type</code> is <code>null</code>
+   */
+  public static boolean deleteQuietly(Path path, String type) {
+    Validate.notNull(path, PathUtils.INVALID_NULL_PATH);
+    Validate.notNull(type, "invalid null type");
+    final File file = path.toFile();
+
+    if (FileUtils.deleteQuietly(file)) {
+      SecurityLogger.audit("{} file {} deleted", StringUtils.capitalize(type), file);
+      return true;
+    }
+    SecurityLogger.audit("Failed to delete {} file {}", StringUtils.lowerCase(type), file);
+    return false;
+  }
+
+  /**
+   * Quietly deletes the specified file or recursively cleans up the specified directory by removing
+   * all empty sub-directories. The provided directory is not deleted even if it becomes empty.
+   *
+   * @param path the file ot directory to be cleaned
+   * @param report the report where to record warning messages if unable to delete the file or
+   *     directories
+   * @return <code>true</code> if the file was deleted or the directory is empty, <code>false</code>
+   *     otherwise
+   * @throws IllegalArgumentException if <code>path</code> or <code>report</code> is <code>null
+   * </code>
+   */
+  public static boolean cleanQuietly(Path path, MigrationReport report) {
+    Validate.notNull(path, PathUtils.INVALID_NULL_PATH);
+    Validate.notNull(report, "invalid null report");
+    final File file = path.toFile();
+
+    if (file.isDirectory()) {
+      return PathUtils.cleanQuietly(file, report);
+    }
+    if (FileUtils.deleteQuietly(file)) {
+      SecurityLogger.audit("File {} deleted", file);
+      return true;
+    }
```

Comment:
:question: Do we need to audit on failure?

Reply:
Yes